### ISSUE TITLE
WebOS - Audio is out of sync

### ISSUE THREAD

**Issue by ivonovak** [xbmc/xbmc#24193]
### Describe the bug
Live Stream Video or MKV files with audio AAC codecs is out of sync. 

Change : last build from 05/12/20023: mkv with aac codec from local storage (nfs) plays in sync

Stream Video, live stream from TVHeadend [ maybe other clients also ] or MKV (H264,H265) files with audio AAC codecs is out of sync., 
but :
Stream Video , live stream from TVHeadend or MKV Files (H264,H265) with AC3, eAC3, DTS codec plays in sync

It is not possible to move the audio to get AV in sync. 
TV : LG OLEG65C, webOS 7
Audio Paththroug : ON 
AC3 transcoding : ON

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

### Debuglog

The debuglog can be found here:
https://paste.kodi.tv/gahukeyogo.kodi

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP
 - [ x ] WebOS

 - Operating system version/name:
 - Kodi version:
WebOS7, Latest KODI build

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by ivonovak** [comment]
Debug log is attached.

**Comment by ivonovak** [comment]
> On my side I don't have sync issues with Opus audio either. As long as passthrough + AC3 transcoding are enabled.
> 
> Didn't try AAC but I'd expect it to behave the same. So maybe it's related to live video.

Yes, it's related to live video. File with AAC codec plays correctly

**Comment by Zibri** [comment]
the problem improves if I set audio passthrough in audio settings. but there is still some delay (less)
also passthrough should be enabled by defalt, let the tv decide how to handle 2.0 or 5.1 streams.
for example, sometimes I have a logitech 5.1 system connected, sometimes I don't.
just pass the stream as it is to the tv so if the speaker setup is "internal speaker" it will be converted to normal stereo... if there is a surround connected, it will be sent as is to it (I enables passthrough also on the tv itself)

**Comment by tuzidaguai** [comment]
First of all, please forgive me for not posting on the https://forum.kodi.tv/showthread.php?tid=368381 topic, for some reason I can't pass the captcha and can't register the account correctly. If I'm posting here and doing something wrong, I'll start by saying I'm sorry.

I don't know English and a large part of the article is translated online through Microsoft.

I may have found some way to avoid the audio-video being out of sync.
My software and hardware environment. LG CX (WebOS 5.5.0), AVR, via HDMI cable, using EARC connection.
kodi 21 rc1，iptv simple client。
Audio Paththroug : ON
AC3 transcoding : ON
Live Stream Video/audio：Multicast,4k，fhd，h265,h264，sdr，hlg，aac，mp2;
addr:rtp://xxx.xxx.xxx.xxx:yyyy
1 Download and install the inputstream.ffmpegdirect add-on correctly in Kodi, this add-on is compiled by satgit62 and cannot be used without compiling on the webos platform.
https://github.com/satgit62/pvr.hts-tvhead-client-on-LG-webOS/blob/main/inputstream.ffmpegdirect.zip

2 Edit the M3U file according to the following rules and re-upload it to the IPTV Simple Client add-on. In order to ensure that the m3u file loads correctly, you can press the switch enable and disable in the add-on several times.

#KODIPROP:inputstream=inputstream.ffmpegdirect
#EXTINF:-1,channel1-4k
rtp://xxx.xxx.xxx.xxx:yyyy
#KODIPROP:inputstream=inputstream.ffmpegdirect
#EXTINF:-1,channel2-sdr
rtp://xxx.xxx.xxx.xxx:yyyy
#KODIPROP:inputstream=inputstream.ffmpegdirect
#EXTINF:-1,channel3-hd
rtp://xxx.xxx.xxx.xxx:yyyy
#KODIPROP:inputstream=inputstream.ffmpegdirect
#EXTINF:-1,channel4-hlg hdr
rtp://xxx.xxx.xxx.xxx:yyyy

3 KODI main interface -- > TV, enter the channel list, use LG remote control to select a channel and play it. After waiting for the sound and image to play normally, press the remote control scroll wheel, and the playback interface of Kodi will appear, please refer to the picture I uploaded to make sure that the Kodi interface is in the correct playback environment. If it's incorrect, quit the current channel and replay the channel until it's correct.
![1](https://github.com/xbmc/xbmc/assets/162652279/1ceb99b7-b375-4fe3-b4c2-f375056b1ef0)
![2](https://github.com/xbmc/xbmc/assets/162652279/6a22d50a-1ea6-4155-9c4f-94a1cee986f1)

**Comment by alexhass** [comment]
I just installed Omega 21.0 final on my LG and see the same issue on all HD streams like ARD/ZDF/ONE/KikaHD and so on... everthing that is HD has stuttering video. The sound seems to be fine... but the picture stutters. No way usable this way. All other non-HD sources like SAT1/RTL are streaming fine.

**Comment by darkcheater** [comment]
Since 21.0. was released now, could someone please tell me how to fix or mitigate the audio sync issues?
They are still there for me after upgrading to 21.0.
Thanks!

**Comment by tuzidaguai** [comment]
Adding tags to m3u files may solve the problem that the audio and video of the live stream are out of sync

This article uses an online translation, and the expression is not very clear.

My environment, LG CX, WebOS 5.5, Kodi 21.0 rc2 version, make sure IPTV Simple Clinet and ffmpegdirect add-ons are enabled. In addition, Kodi and all  add-ons settings remain default.
You can also use the stable version of Kodi 21.0, but you need to download the ffmpegdirect  add-ons by yourself, and the download address is https://raw.githubusercontent.com/satgit62/pvr.hts-tvhead-client-on-LG-webOS/main/inputstream.ffmpegdirect.zip

Edit the m3u file in the following format and upload the edited file in your own way.
#KODIPROP:inputstream=inputstream.ffmpegdirect
rtp://xx.xx.xx.xx:xxxx
#KODIPROP:inputstream=inputstream.ffmpegdirect
rtsp://xx.xx.xx.xx/.......xx.smil
#KODIPROP:inputstream=inputstream.ffmpegdirect
http://xx.xx.xx.xx/.......xx.m3u8
#KODIPROP:inputstream=inputstream.ffmpegdirect
http://xx.xx.xx.xx/......./xx.php?id=ch1

Once you've done the above, use Kodi to open the channel of the M3U you just uploaded. Note that when opening a channel, two buttons must appear to force the use of the ffmpegdirect  add-ons, as shown in pic 1. If one button appears (pic 2), you need to close the channel and re-open until two buttons appear.

Finally, with ffmpegdirect, there is an occasional problem, see pic 3, which buffers for a few seconds, and then plays normally and no longer buffers. But I didn't find the relevant setting to cancel it, and it doesn't come up every time.

![1](https://github.com/xbmc/xbmc/assets/162652279/09395551-56e3-4ce0-b4d3-3ee95c5fb128)
![2](https://github.com/xbmc/xbmc/assets/162652279/845f6c71-f9d4-453c-a5de-27bf33f71403)
![3](https://github.com/xbmc/xbmc/assets/162652279/30439472-538f-41c0-bd4b-b7201fca124b)

**Comment by digglife** [comment]
still an issue. Could any developer please give a hint where the problem could be, so I can give it a try?

**Comment by sorryusernameisalreadytaken** [comment]
Is it better now with the latest release?

**Comment by Juancafr** [comment]
Is there still no solution?

**Comment by lundman** [comment]
Apologies for being late, but `sundermann closed this as completed [#26988]`.. just here to report this works great. I tried the `org.xbmc.kodi_20250731-e453465f-webos-videoplayer_arm.ipk` version and have everything I need working, syncing is excellent, rar support etc etc.  Great job!
